src/Uno.UI.Tests/Windows_UI_Xaml/Given_Implicit_Style.cs

```diff
@@ -23,5 +39,270 @@ public void When_Implicit_Style_In_Application_Merged()
+			button.Style = style;
+			Assert.AreEqual(new Thickness(), button.Padding); //Default style not applied yet, we haven't been added to the visual tree.
+		}
+
+		[TestMethod]
+		public void When_Newly_Created_Framework_Type_ApplyTemplate()
+		{
+			var button = new Button() { /*No Content - note that if we had content, a default template would be created*/ };
+
+			button.ApplyTemplate();
+
+			Assert.AreEqual(new Thickness(), button.Padding);
+			Assert.IsNull(button.Template);
+		}
+
+		[TestMethod]
+		public void When_Implicit_Style_And_Programmatic_Explicit_Style()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var testControl = new Test_Control();
+
+			Assert.AreEqual("AppLevelImplicit", testControl.TestRadioButton.Tag);
+
+			var style = app.Resources["ExplicitRadioButtonStyle1"] as Style;
+			Assert.IsNotNull(style);
+			testControl.TestRadioButton.Style = style;
+			Assert.AreEqual(Colors.Moccasin, (testControl.TestRadioButton.Background as SolidColorBrush).Color);
+			Assert.IsNull(testControl.TestRadioButton.Tag);
+		}
+
+		[TestMethod]
+		public void When_Inheriting_From_Type_With_Implicit_Style()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var testControl = new Test_Control();
+
+			app.HostView.Children.Add(testControl);
+
+			Assert.AreEqual("InnerTreeImplicit", testControl.TestRadioButton.Tag);
+			Assert.AreEqual(typeof(RadioButton), testControl.StylesTestRadioButton.ExposedKey);
+			Assert.IsNull(testControl.StylesTestRadioButton.Tag); //Inherited type doesn't use implicit style for base type
+		}
+
+		[TestMethod]
+		public void When_From_External_Library_And_CodeBehind()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var extControl = new MyExtControl();
+
+			app.HostView.Children.Add(extControl);
+
+			Assert.AreEqual("FromDefaultStyle", extControl.MyTag);
+		}
+
+		[TestMethod]
+		public void When_From_External_Library_And_Xaml()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var testPage = new Test_Page();
+
+			app.HostView.Children.Add(testPage);
+
+			Assert.AreEqual("FromDefaultStyle", testPage.MyExtControl.MyTag);
+			Assert.AreEqual("FromLocalImplicit", testPage.MyExtControl.MyTag2);
+		}
+
+		[TestMethod]
+		public void When_Inherited_Property()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var testPage = new Test_Page();
+
+			app.HostView.Children.Add(testPage);
+
+			testPage.Measure(new Size(1000, 1000));
+
+			var accentColor = (Color)app.Resources["AccentColorAlias"];
+			AssertEx.AssertHasColor(testPage.OuterHyperlinkButton.Foreground, accentColor);
+			AssertEx.AssertHasColor(testPage.InnerHyperlinkButton.Foreground, accentColor);
+		}
+
+		[TestMethod]
+		public void Verify_Framework_DefaultStyleKeys()
+		{
+			var keyProp = typeof(Control)
+#if NETFX_CORE
+				.GetTypeInfo()
+				.GetDeclaredProperty("DefaultStyleKey");
+#else
+				.GetProperty("DefaultStyleKey", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make sure that this accessibility bypass is safe here.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)